weldx/visualization.py

```diff
@@ -83,3 +324,1128 @@ def set_axes_equal(axes):
+        self._lcs = lcs
+        self._color = color
+
+        self._vectors = k3d.vectors(
+            origins=[coordinates for _ in range(3)],
+            vectors=orientation.transpose(),
+            colors=[[0xFF0000, 0xFF0000], [0x00FF00, 0x00FF00], [0x0000FF, 0x0000FF]],
+            labels=[],
+            label_size=1.5,
+        )
+        self._vectors.visible = show_vectors
+
+        self._label = None
+        if name is not None:
+            self._label = k3d.text(
+                text=name,
+                position=coordinates + 0.05,
+                color=self._color,
+                size=1,
+                label_box=False,
+            )
+
+        self._trace = k3d.line(
+            np.array(lcs.coordinates.values, dtype="float32"),
+            shader="simple",
+            width=0.05,
+            color=color,
+        )
+        self._trace.visible = show_trace
+
+        self.origin = platonic.Octahedron(size=0.1).mesh
+        self.origin.color = color
+        self.origin.model_matrix = self._create_model_matrix(coordinates, orientation)
+        self.origin.visible = show_origin
+
+        if plot is not None:
+            plot += self._vectors
+            plot += self._trace
+            plot += self.origin
+            if self._label is not None:
+                plot += self._label
+
+    @staticmethod
+    def _create_model_matrix(
+        coordinates: np.ndarray, orientation: np.ndarray
+    ) -> np.ndarray:
+        """Create the model matrix from an orientation and coordinates.
+
+        Parameters
+        ----------
+        coordinates : numpy.ndarray
+            The coordinates of the origin
+        orientation : numpy.ndarray
+            The orientation of the coordinate system
+
+        Returns
+        -------
+        numpy.ndarray :
+            The model matrix
+
+        """
+        model_matrix = np.eye(4, dtype="float32")
+        model_matrix[:3, :3] = orientation
+        model_matrix[:3, 3] = coordinates
+        return model_matrix
+
+    @staticmethod
+    def _get_coordinates_and_orientation(lcs, index: int = 0):
+        """Get the coordinates and orientation of a coordinate system.
+
+        Parameters
+        ----------
+        lcs : weldx.LocalCoordinateSystem
+            The coordinate system
+        index : int
+            If the coordinate system is time dependent, the passed value is the index
+            of the values that should be returned
+
+        Returns
+        -------
+        coordinates : numpy.ndarray
+            The coordinates
+        orientation : numpy.ndarray
+            The orientation
+
+        """
+        coordinates = np.array(lcs.coordinates.values, dtype="float32")
+        orientation = np.array(lcs.orientation.values, dtype="float32")
+        if coordinates.ndim > 1:
+            coordinates = coordinates[index]
+
+        if orientation.ndim > 2:
+            orientation = orientation[index]
```

Comment:
How about some xarray indexing? 🧙 (I think this should work 😉) 
```suggestion
        coordinates = lcs.coordinates.isel(time=index, missing_dims="ignore").values.astype("float32")
        orientation = lcs.orientation.isel(time=index, missing_dims="ignore").values.astype("float32")
```

Reply:
Seems to work :)